yarpl/include/yarpl/flowable/CancelingSubscriber.h

```diff
@@ -13,16 +13,21 @@ namespace flowable {
  * A Subscriber that always cancels the subscription passed to it.
  */
 template <typename T>
-class CancelingSubscriber final : public InternalSubscriber<T> {
+class CancelingSubscriber final : public BaseSubscriber<T> {
  public:
-  void onSubscribe(
-      yarpl::Reference<yarpl::flowable::Subscription> sub) override {
-    sub->cancel();
+  void onSubscribeImpl() override {
+    this->cancel();
   }
 
-  void onNext(T) override {
+  void onNextImpl(T) override {
```

Comment:
Would it be possible to make these private or protected?

Reply:
There wouldn't be much point; just cast to a BaseSubscriber (which is always going to be valid) and they're public again.

Reply:
I mean why they are public in the BaseSubscriber at all?